DuckDuckGo/AutofillLoginListViewModel.swift

```diff
@@ -195,12 +209,34 @@ final class AutofillLoginListViewModel: ObservableObject {
         }
     }
     
-    private func delete(_ account: SecureVaultModels.WebsiteAccount) {
+    @discardableResult
+    private func delete(_ account: SecureVaultModels.WebsiteAccount) -> Bool {
         guard let secureVault = try? SecureVaultFactory.default.makeVault(errorReporter: SecureVaultErrorReporter.shared),
-              let accountID = account.id else { return }
+              let accountID = account.id else { return false }
         
         do {
+            cachedDeletedCredentials = try secureVault.websiteCredentialsFor(accountId: accountID)
             try secureVault.deleteWebsiteCredentialsFor(accountId: accountID)
+            return true
+        } catch {
+            Pixel.fire(pixel: .secureVaultError)
+            return false
+        }
+    }
+    
+    private func undelete(_ account: SecureVaultModels.WebsiteCredentials) {
```

Comment:
How do you feel about integrating this with [`UndoManager`](https://developer.apple.com/documentation/foundation/undomanager)? That might be a bit of a stretch goal since we have our own custom undo UI though, so I don't consider it a blocker to leave this for now.

Reply:
I'm struggling to think of a benefit that makes it worth it tbh (i.e. if this was going to be the totality of our undoing support on iOS or possibly even macOS, I think not using undo manager would be the way to go long term.). I could be convinced if we expect to have much more undoing and/or redoing in future